R/subset_county_polygons.R

```diff
@@ -0,0 +1,39 @@
+#' subset_county_polygons
+#'
+#' Returns county polygons from: Siczewicz, Peter. U.S. Historical Counties 
+#' (Generalized .001 deg) on the basis of year of interest
+#' Dataset is subset on basis of area of interest
+#' 
+#' @param areas chr, vector of codes indicating HUCs, counties, states, regions, aquifers, etc. 
+#' @param area.column chr, defines which column to use to specify area
+#' @param year int, the year of interest to be mapped (defines historical basis for counties)
+#' 
+#' @export
+#' 
+#' @importFrom tidyr gather_
+#' 
+#' @examples 
+#' areas <- "Delaware" # should yield 3 counties
+#' areas <- "Maine"# should yield 16 counties
+#' areas <- "New Hampshire"# should yield 10 counties
+#' area.column <- "STATE_TERR"
+#' year <- 2010
+#' subset_county_polygons(areas, area.column, year)
+
+
+subset_county_polygons <- function(areas=NA, area.column, year){
+  
+  test.date <- as.Date(paste0(year,"-12-31"))# year of interest converted to date
+  now.date <-format(Sys.Date(), "%Y-%m-%d")
+  # select area of interest
+  hc_sub <- histCounties[which(histCounties@data[,area.column] == areas),]
+  
+  # substitute todays date for the last date of the histCounties dataset (2000-12-31)
+  hc_sub$END_DATE[hc_sub$END_DATE==max(hc_sub$END_DATE)] <- now.date
+  # assuming the counties defined in 2000-12-31 are the same now, use todays 
+  # date to make sure the selection for contemporaneous counties works  
+  hc_sub <- hc_sub[hc_sub$END_DATE >= test.date,]# for year of interest, get the counties 
```

Comment:
I think you'd have to test both end and start dates if those dates for a given county overlapped. Assuming they don't, (that is, except for the first start date, it seems the start date for a county equals the end date of the same county from a previous time period). In which case, I think it's sufficient to check only that the end date is greater than the year you want to look at but I could be wrong. However, we do need to handle the possibility of getting two end dates greater than the year of interest. Say, year of interest is 2000 and counties changed in 2005 and 2010 - we'd want to select the oldest end date. Want me to make those edits before merging?